File: java-todo-app/src/main/java/com/example/todo/application/service/TodoService.java
package com.example.todo.application.service;

import com.example.todo.application.dto.TodoRequest;
import com.example.todo.application.dto.TodoResponse;

import java.util.List;

public interface TodoService {
    List<TodoResponse> getTodos();

    TodoResponse saveTodo(TodoRequest todoRequest);

    void deleteTodo(Long id);
}

File: kotlin-todo-app/src/main/kotlin/com/example/todo/application/service/TodoService.kt
package com.example.todo.application.service

import com.example.todo.application.dto.TodoRequest
import com.example.todo.application.dto.TodoResponse

interface TodoService {
    fun getTodos(): List<TodoResponse>
    fun saveTodo(todoRequest: TodoRequest): TodoResponse
    fun deleteTodo(id: Long)
}
